Framework/Source/API/ConstantBuffer.cpp

```diff
@@ -88,148 +88,93 @@ namespace Falcor
         return mpCbv;
     }
 
-    bool ConstantBuffer::getGuiWidgetFromType(Gui* pGui, ReflectionBasicType::Type type, uint8_t* data, const std::string& name)
+    bool ConstantBuffer::getGuiWidgetFromType(Gui* pGui, ReflectionBasicType::Type type, size_t offset, const std::string& name)
     {
-        std::string displayName = name;
-        displayName.resize(name.size() + 3, 0);
-        size_t offset = 0;
         unsigned displayIndex = 0;
         bool returnValue = false;
 
-
-#define update_array_index_name() displayName[displayName.size() - 3] = '['; \
-        displayName[displayName.size() - 2] = '0' + displayIndex; \
-        displayName[displayName.size() - 1] = ']'; \
-        displayIndex++
 #define concatStrings_(a, b) a##b
 #define concatStrings(a, b) concatStrings_(a, b)
 #define to_gui_widget(widgetName, baseType) \
-        returnValue |= pGui-> concatStrings(add, widgetName)(displayName.c_str(), *reinterpret_cast<baseType*>(data + offset)); \
+        returnValue |= pGui-> concatStrings(add, widgetName)(name.c_str(), *reinterpret_cast<baseType*>(mData.data() + offset)); \
```

Comment:
This can just be a basic assignment since this macro should only be called once in the switch-case